lib/hive/controller/android.rb

```diff
@@ -46,25 +50,23 @@ def get_connected_devices
             }
           end
 
-          attached_devices = device_info.collect do |physical_device|
+          # attached_devices is either fully set here (if there is no Hivemind connection) 
+          # or in select_devices (if there is a Hivemind connection), not both
+          @attached_devices = device_info.collect do |physical_device|
             self.create_device(physical_device)
           end
         end
 
-        Hive.logger.info(attached_devices)
         connected_devices
       end
 
       def select_devices(connected_devices)
         # select devices that we want to poll
-        attached_devices = []
         to_poll = []
+        @attached_devices || []
```

Comment:
Shouldn't that be '||= []' otherwise you're not assigning anything to `@attached_devices` if it hasn't been set